rfc/rfc-0003-intermediate-value-crds.md

```diff
@@ -1,3 +1,140 @@
+  template:
+    apiVersion: source.toolkit.fluxcd.io/v1beta1
+    kind: GitRepository
+    metadata:
+      name: $(workload.name)$-source
+    spec:
+      interval: 5m
+      url: $(workload.git.url)$
+      ref: $(workload.git.ref)$
+```
+
+```yaml
+## Resulting proper GitResource
+apiVersion: source.toolkit.fluxcd.io/v1beta1
+kind: GitRepository
+metadata:
+  creationTimestamp: "2021-03-02T22:25:30Z"
+  finalizers:
+  - finalizers.fluxcd.io
+  generation: 22
+  labels:
+    experimental.kontinue.io/digest: 3a2531a9382654b15865220e9b51f8d4
+  name: example-app
+  namespace: default
+  ownerReferences:
+  - apiVersion: experimental.kontinue.io/v1
+    blockOwnerDeletion: true
+    controller: true
+    kind: Workload
+    name: example-app
+    uid: 82698cb8-2c1a-4c55-8742-25b13b39521a
+  resourceVersion: "53158"
+  uid: 0d5b2a20-adb9-41f7-b61c-ee434596416e
+spec:
+  gitImplementation: go-git
+  interval: 1m0s
+  ref:
+    branch: master
+  secretRef:
+    name: gitlab-credentials
+  timeout: 20s
+  url: https://gitlab.eng.vmware.com/tanzu-delivery-pipeline/example-app
+status:
+  artifact:
+    checksum: fb5edd0cd4ab245791af41aea8ab8654a835ee87
+    lastUpdateTime: "2021-03-02T22:48:03Z"
+    path: gitrepository/default/example-app/115fc159352e68d4b4b17fd358924faa9fce1d22.tar.gz
+    revision: master/115fc159352e68d4b4b17fd358924faa9fce1d22
+    url: http://source-controller.gitops-toolkit.svc.cluster.local./gitrepository/default/example-app/115fc159352e68d4b4b17fd358924faa9fce1d22.tar.gz
+  conditions:
+  - lastTransitionTime: "2021-03-02T22:48:03Z"
+    message: 'Fetched revision: master/115fc159352e68d4b4b17fd358924faa9fce1d22'
+    reason: GitOperationSucceed
+    status: "True"
+    type: Ready
+  observedGeneration: 22
+  url: http://source-controller.gitops-toolkit.svc.cluster.local./gitrepository/default/example-app/latest.tar.gz
+```
+
+```yaml
+## Example Intermediate Value CRD where error can be observed
+apiVersion: kontinue.io/v1alpha1
+kind: IntermediateValue
+metadata:
+  name: git-template
+  labels:
+    kontinue.io/workload-name: petclinic
+    kontinue.io/cluster-supply-chain-name: responsible-ops
+    kontinue.io/component-name: source-provider
+    kontinue.io/cluster-template-name: git-template
+  ownerReferences:
+    - apiVersion: kontinue.io/v1alpha1
+      kind: Workload
+      name: petclinic
+spec:
+  url: http://source-controller.gitops-toolkit.svc.cluster.local./gitrepository/default/example-app/115fc159352e68d4b4b17fd358924faa9fce1d22.tar.gz
+  revision: http://source-controller.gitops-toolkit.svc.cluster.local./gitrepository/default/example-app/115fc159352e68d4b4b17fd358924faa9fce1d22.tar.gz
+```
+
+## Rationale and Alternatives
+
+- Create an IntermediateSourceValue, IntermediateImageValue, IntermediateConfigValue. This is
+  a small interation on IntermediateValue CRD.
+- Store the information on the workload status. This would mean creating fewer objects in etcd
```

Comment:
I think we'll end up wanting to store this information in Workload status regardless, so that we can provide traceability for active (and maybe even recent) artifacts. Attempting to trace artifacts by following the created resources would be significantly slower.